### DISCUSSION TITLE
Action to close visible notifications only

### DISCUSSION THREAD

**Discussion by ouuan** [dunst-project/dunst#1346] [Ideas]
The `close_all` action closes all notifications regardless of their visibility. When there are more than `notification_limit` notifications, I often want to close only the visible ones, so that I can read the hidden ones before closing them.

**Comment by zappolowski** [comment]
For a similar use case (selectively closing some notifications) I've implemented an new DBus method to list currently displayed notifications (see #1348). Using this one (without any filter) should do what you need, or?

**Reply by bynect** [reply]
I don't think this is easily portable to click actions. Maybe with a custom script. But adding a `close_displayed` doesn't sound too unreasonable.

**Reply by zappolowski** [reply]
I misinterpreted the term `action` to be the context of `dunstctl` which has the `close-all` action already, but there's also `close_all` as an action on notifications itself, which was probably meant.

Sure, adding `close_displayed` there as well is a possible solution. I'm just a bit concerned about adding too many "custom" actions, when similar functionality is already present somewhere else (DBus/dunstctl in this case).

E.g. the `close_all` action could just be a script which calls `dunstctl close-all`. When the DBus interface becomes more expressive (e.g. `dunstctl close-all --displayed`) this would be immediately available for mouse actions as well (or just with minor adjustments to a script).

**Reply by bynect** [reply]
> E.g. the close_all action could just be a script which calls dunstctl close-all. When the DBus interface becomes more expressive (e.g. dunstctl close-all --displayed) this would be immediately available for mouse actions as well (or just with minor adjustments to a script).

doesn't sound bad.

>  I'm just a bit concerned about adding too many "custom" actions, when similar functionality is already present somewhere else (DBus/dunstctl in this case).

Maybe we can add the possibility (in v2) for users to create custom actions (without the need to create a script).
